dll/cpl/input/settings_page.c

```diff
@@ -619,23 +600,19 @@ OnNotifySettingsPage(HWND hwndDlg, LPARAM lParam)
 
         case PSN_APPLY:
         {
-            BOOL bRebootNeeded = IsRebootNeeded();
-
-            /* Write Input Methods list to registry */
-            if (InputList_Process() && bRebootNeeded)
+            /* If an HKL is IME, then reboot is needed */
+            INPUT_LIST_NODE *pNode;
+            for (pNode = InputList_GetFirst(); pNode != NULL; pNode = pNode->pNext)
             {
-                /* Needs reboot */
-                WCHAR szNeedsReboot[128], szLanguage[64];
-                LoadStringW(hApplet, IDS_REBOOT_NOW, szNeedsReboot, _countof(szNeedsReboot));
-                LoadStringW(hApplet, IDS_LANGUAGE, szLanguage, _countof(szLanguage));
-
-                if (MessageBoxW(hwndDlg, szNeedsReboot, szLanguage,
-                                MB_ICONINFORMATION | MB_YESNOCANCEL) == IDYES)
+                if (IS_IME_HKL(pNode->hkl)) /* IME? */
                 {
-                    EnableProcessPrivileges(SE_SHUTDOWN_NAME, TRUE);
-                    ExitWindowsEx(EWX_REBOOT | EWX_FORCE, 0);
+                    g_bRebootNeeded = TRUE;
+                    break;
                 }
             }
+
+            /* Write Input Methods list to registry */
+            InputList_Process();
```

Comment:
Shouldn't reboot be needed only if this function returns TRUE?

Reply:
OK.